pkgs/applications/networking/cluster/k3s/builder.nix

```
@@ -261,67 +266,71 @@ let
   # derivation when we've built all the binaries, but haven't bundled them in
   # with generated bindata yet.
 
-  k3sServer = buildGoModule {
-    pname = "k3s-server";
-    version = k3sVersion;
-
-    src = k3sRepo;
-    vendorHash = k3sVendorHash;
-
-    nativeBuildInputs = [ pkg-config ];
-    buildInputs = [
-      libseccomp
-      sqlite.dev
-    ];
-
-    subPackages = [ "cmd/server" ];
-    ldflags = versionldflags;
-
-    tags = [
-      "ctrd"
-      "libsqlite3"
-      "linux"
-    ];
-
-    # create the multicall symlinks for k3s
-    postInstall = ''
-      mv $out/bin/server $out/bin/k3s
-      pushd $out
-      # taken verbatim from https://github.com/k3s-io/k3s/blob/v1.23.3%2Bk3s1/scripts/build#L105-L113
-      ln -s k3s ./bin/containerd
-      ln -s k3s ./bin/crictl
-      ln -s k3s ./bin/ctr
-      ln -s k3s ./bin/k3s-agent
-      ln -s k3s ./bin/k3s-certificate
-      ln -s k3s ./bin/k3s-completion
-      ln -s k3s ./bin/k3s-etcd-snapshot
-      ln -s k3s ./bin/k3s-secrets-encrypt
-      ln -s k3s ./bin/k3s-server
-      ln -s k3s ./bin/k3s-token
-      ln -s k3s ./bin/kubectl
-      popd
-    '';
-
-    meta = baseMeta // {
-      description = "Various binaries that get packaged into the final k3s binary";
-    };
-  };
+  k3sBin =
```

```suggestion
  k3sBin =
```
> k3sServer to k3sBin

`bin` meaning `binary`. I found this rename weird. I don't have a better suggestion atm.

---

I think the name is more fitting since this builds the k3s binary that can be invoked as server, agent, kubectl and so on. The former `k3sServer` isn't ideal, for example a user would have to override `k3sServer` to modify the build of the k3s agent.

However, the result of the derivation looks like multiple binaries due to all the symlinks that are created, what do you think about `k3sBinaries`?

---

Maybe k3sBundle?

---

That's fine for me